go/control/api/api.go

```diff
@@ -124,6 +124,12 @@ type IdentityStatus struct {
 
 // RegistrationStatus is the node registration status.
 type RegistrationStatus struct {
+	// Successful is true if the current registration has been successful.
+	Successful bool `json:"successful"`
```

Comment:
Hmm, I'm not sure...  `Registered` in your case is basically the same as checking if `Descriptor` is not `nil`.  Adding a timestamp for the last registration attempt is a good idea, I'll do that.  And I need to add better comments for the existing fields.

Reply:
How about this, I:
- add the field `LastAttempt` (timestamp)
- rename `Successful` to `LastAttemptSuccessful`
- rename `Error` to `LastAtemptError`

This will make it more clear what the new fields actually mean.  I think that having the bool and the error separate is still useful for quickly checking if the node has problems (especially when using automation) and once I change the status updating as you've suggested above, there's really no way that the bool and the error could be out of sync :)

Reply:
Looks good, have no better suggestions. Putting this under `last_attempt` with fields `timestamp`, `successful` and `error_message` would probably be an overkill.